src/rgw/s3select_csv_parser.h

```diff
@@ -0,0 +1,332 @@
+#include <iostream>
+
+#include <boost/mpl/vector/vector30.hpp>
+// back-end
+#include <boost/msm/back/state_machine.hpp>
```

Comment:
seems like a good idea to use state_machine here

Reply:
it also perform quite well (there still place for improvements)